### PR TITLE
feat(avatar): add avatar group

### PR DESCRIPTION
**Pull request by nkrantz**
Adds Avatar Group component to `@twilio-paste/avatar`

### CODE REVIEW

File: packages/paste-core/components/avatar/__tests__/avatar.test.tsx [hunk 1]
```diff
@@ -105,6 +106,15 @@ describe('Avatar', () => {
...
     });
+
+    describe('getGroupSpacing', () => {
+      it('should return the correct space token', () => {
+        expect(getGroupSpacing('sizeIcon30', 'user')).toEqual('spaceNegative10');
+        expect(getGroupSpacing('sizeIcon100', 'user')).toEqual('spaceNegative40');
+        expect(getGroupSpacing('sizeIcon30', 'entity')).toEqual('spaceNegative10');
+        expect(getGroupSpacing('sizeIcon100', 'entity')).toEqual('spaceNegative20');
+      });
+    });
```
Comment (SiTaggart) [inline packages/paste-core/components/avatar/__tests__/avatar.test.tsx:117]: praise: nice to see tests!

File: packages/paste-core/components/avatar/src/utils.ts [hunk 2]
```diff
@@ -119,3 +122,47 @@ export const getComputedTokenNames = (
...
+        return 'spaceNegative20';
+      case 'sizeIcon80':
+      case 'sizeIcon90':
+        return 'spaceNegative30';
+      case 'sizeIcon100':
+        return 'spaceNegative40';
+      case 'sizeIcon110':
+        return 'spaceNegative50';
+    }
+  } else if (variant === 'entity') {
```
Comment (SiTaggart) [inline packages/paste-core/components/avatar/src/utils.ts:150]: suggestion: you might not need this else if you provide a default case with the next switch statement, as it looks like it's just `spaceNegative20` if all else fails

File: packages/paste-website/src/pages/components/avatar/index.mdx [hunk 3]
```diff
@@ -154,42 +177,29 @@ Provide the Avatar with an `icon` prop to display an icon. You must import the i
...
-  <Avatar variant="entity" name="entity example" icon={BusinessIcon} />
+  <Avatar size="sizeIcon30" name="Dakota Powers" icon={UserIcon} />
+  <Avatar size="sizeIcon40" name="Maria Ardovini" icon={UserIcon} />
+  <Avatar size="sizeIcon50" name="Imani Nakai" icon={UserIcon} />
+  <Avatar size="sizeIcon60" name="Merlyn Lund" icon={UserIcon} />
+  <Avatar size="sizeIcon70" name="Twilio Engage" variant="entity" icon={ProductEngageIcon} />
+  <Avatar size="sizeIcon80" name="Twilio Flex" variant="entity" icon={ProductFlexIcon} />
+  <Avatar size="sizeIcon90" name="Twilio Voice" variant="entity" icon={ProductVoiceIcon} />
+  <Avatar size="sizeIcon100" name="Twilio Studio" variant="entity" icon={ProductStudioIcon} />
+  <Avatar size="sizeIcon110" name="Twilio Interconnect" variant="entity" icon={ProductInterconnectIcon} />
```
Comment (SiTaggart) [inline packages/paste-website/src/pages/components/avatar/index.mdx:196]: praise: loving these new examples 
image

File: packages/paste-website/src/pages/components/avatar/index.mdx [hunk 4]
```diff
@@ -55,51 +62,67 @@ export const getStaticProps = async () => {
...
-    no image or icon is passed will the Avatar display initials.
-  </CalloutText>
-</Callout>
+Avatar sizing works similarly to [Paste Icon](/components/icons/usage-guidelines#resizing-icons) sizing. Pass any of the icon size tokens (`sizeIcon10`, `sizeIcon20`, `sizeIcon30`, etc.) to Avatar’s `size` prop.
+
+**Note:** using sizes 10 and 20 isn’t recommended for displaying initials within an Avatar, as these small sizes may cut off certain wide characters, such as “W”.
+
+### Accessibility
+
+The `name` prop is required on all instances of the Avatar component. This ensures that whether your Avatar is displaying initials, an image, or an icon, the user will be able to access information about what object is being represented by the Avatar. The full string passed to `name` appears as a title upon hovering the Avatar, and is useful to assistive technology users.
```
Comment (SiTaggart) [inline packages/paste-website/src/pages/components/avatar/index.mdx:95]: praise: excellent

File: packages/paste-core/components/avatar/src/AvatarGroup.tsx [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import {Stack} from '@twilio-paste/stack';
+import {isIconSizeTokenProp} from '@twilio-paste/style-props';
+
+import type {AvatarGroupProps, AvatarVariants} from './types';
+import {getGroupSpacing} from './utils';
+
+export const AvatarContext = React.createContext<Omit<AvatarGroupProps, 'children'>>({} as AvatarGroupProps);
```
Comment (TheSisb) [inline packages/paste-core/components/avatar/src/AvatarGroup.tsx:9]: suggestion: let's call this `AvatarGroupContext` for more clarity

File: packages/paste-core/components/avatar/src/index.tsx [hunk 6]
```diff
@@ -152,3 +155,4 @@ Avatar.propTypes = {
 };
 
 export {Avatar};
+export {AvatarGroup} from './AvatarGroup';
```
Comment (TheSisb) [inline packages/paste-core/components/avatar/src/index.tsx:158]: suggestion: Let's rename this file to `Avatar.tsx` and add a new `index.tsx` file with these exports. It's the new pattern we've been mostly following and makes this file more clear.

File: packages/paste-website/src/pages/components/avatar/index.mdx (hunk 4 above)
Comment (TheSisb) [reply packages/paste-website/src/pages/components/avatar/index.mdx:95]: +1

**Review by SiTaggart — APPROVED** [review]

**Review by loreina — APPROVED** [review]

**Review by loreina — APPROVED** [review]

**Pull request merged**